test/box-luatest/alter_with_compression_test.lua

```diff
@@ -0,0 +1,125 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+
+local g = t.group("invalid compression type", t.helpers.matrix({
+    engine = {'memtx', 'vinyl'},
+    compression = {'zstd', 'lz4'}
+}))
+
+g.before_all(function(cg)
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:stop()
+end)
+
+g.test_invalid_compression_type_during_space_creation = function(cg)
+    local format = {{
+        name = 'x', type = 'unsigned', compression = cg.params.compression
+    }}
+
+    t.assert_error_msg_content_equals(
+        "Failed to create space 'T': field 1 has unknown compression type",
+        function()
+            cg.server:exec(function(engine, format)
+                return box.schema.space.create('T', {
+                    engine = engine, format = format
+                })
+            end, {cg.params.engine, format})
+        end
+    )
+end
+
+g.before_test('test_invalid_compression_type_during_setting_format', function(cg)
+    cg.server:exec(function(engine)
+        box.schema.space.create('space', {engine = engine})
+    end, {cg.params.engine})
+end)
+
+g.test_invalid_compression_type_during_setting_format = function(cg)
+    local format = {{
+        name = 'x', type = 'unsigned', compression = cg.params.compression
+    }}
+
+    t.assert_error_msg_content_equals(
+        "Can't modify space 'space': field 1 has unknown compression type",
+        function()
+            cg.server:exec(function(format)
+                return box.space.space:format(format)
+            end, {format})
+        end
+    )
+
+    t.assert_error_msg_content_equals(
+        "Can't modify space 'space': field 1 has unknown compression type",
+        function()
+            cg.server:exec(function(format)
+                return box.space.space:alter({format = format})
+            end, {format})
+        end
+    )
```

Comment:
But if we wan't to check error message content it will be more code. Also we should `require` luatest all time. May be it's better to leave it as it is when we check error message?

Reply:
Yes, i don't understant idea first time! Fixed